### ISSUE TITLE
Set the maximum length of a text field

### ISSUE THREAD

**Issue by CarloBu** [sveltia/sveltia-cms#141]
I'm building a strict Website layout where I want to limit the maximum character length for the client.
![image](https://github.com/sveltia/sveltia-cms/assets/145860771/54a50ce6-9ed0-46af-b003-8e7267df8cbe)

By the example above, If the client puts an entire Shakespeare sentence in the title, it will break all the design. It would be great to limit the character count, but I miss this feature in the Sveltia CMS.

would it be possible to implement this feature using regex? For example, yaml could look like this:

```
  - name: test
    label: Test
    folder: src/content/test
    create: true
    fields:
      - label: Name
        name: name
        widget: string
        pattern: ['^.{0,70}$', 'Must not exceed 70 characters']  <- using regex
```
and the UI could show the current character count and maximum count - 65/70 like in the example in Sanity UI.

![image](https://github.com/sveltia/sveltia-cms/assets/145860771/03996be2-109e-4d3e-a938-08105ad50af1)

Going even further could be possible to implement the range:
pattern: ['^.{20,70}$', 'Must not exceed 70 characters']  <- using regex
and to show in the UI:
20/<current character count>/70
20/36/70
but I don't know at what scenario to use the min character limit.

**Comment by kyoshino** [comment]
I’m actually thinking about adding the `min` and `max` options to the String and Text widgets so that you don’t have to use a custom regex.

**Comment by CarloBu** [comment]
that would be even better for DX. 👍 
you could give the nice UX addition by changing the color of the character counter to red when the pattern is not met, and to blue when the character count meets the requirements.

**Comment by kyoshino** [comment]
Yeah, I’m thinking about it. 😆

**Comment by kyoshino** [comment]
Just shipped [v0.27.0](https://github.com/sveltia/sveltia-cms/releases/tag/v0.27.0) with the new options!

**Comment by CarloBu** [comment]
this is so clean from DX side:

  - name: test
    label: Test
    folder: src/content/test
    create: true
    fields:
      - label: Title
        name: title
        widget: string
        minlength: 10
        maxlength: 70

![image](https://github.com/sveltia/sveltia-cms/assets/145860771/65a4ec80-2b36-4dc3-865b-7dc0ccb72528)
